watertrading/code/BuildModel.py

```diff
@@ -0,0 +1,1361 @@
+    for pi in model.s_PI:
+        for t in model.s_T_pi[pi]:
+            NodalSupply[Producer_Wellpad[pi],t] += Producer_Rate[pi]
+    
+    NodalDemand = dict(((c,t),0.0) for c in model.s_CP for t in model.s_T)
+    for ci in model.s_CI:
+        for t in model.s_T_ci[ci]:
+            NodalDemand[Consumer_Wellpad[ci],t] += Consumer_Rate[ci]
+    
+    def p_FP_Trucked_UB_init(model, p, c, t):
+        return min(NodalSupply[p,t], NodalDemand[c,t])
+    """
+    
+    # update for new arc indexing:
+    def p_FP_Trucked_UB_init_V2(model, pi, p, c, t):
+        if tp in model.s_T_pi[pi]:
+            return model.p_ProducerRate[pi]
+        else:
+            return 0
+        
+    model.p_FP_Trucked_UB = Param(
+        model.s_LLT,
+        within=Any,
+        default=None,
+        mutable=True,
+        #initialize=p_FP_Trucked_UB_init,
+        initialize=p_FP_Trucked_UB_init_V2,
+        units=model.model_units["volume_time"],
+        doc="Maximum trucking capacity between nodes [volume_time]",
+    )
+
+    # Producer transport bound
+    for (pi,p,c,t) in model.s_LLT:
+        model.v_FP_Trucked[pi, p, c, t].setub(model.p_FP_Trucked_UB[pi, p, c, t])
+
+    # Midstream transport bound
+    for (mi,p,c,tp,tc) in model.s_LLM:
+        model.v_FM_Trucked[mi, p, c, tp, tc].setub(model.p_MidstreamRate[mi])
+
+    # Supply bound
+    for pi in model.s_PI:
+        for t in model.s_T:
+            if t in model.s_T_pi[pi]:
+                model.v_Supply[pi, t].setub(model.p_ProducerRate[pi])
+            else:
+                model.v_Supply[pi, t].setub(0)
+
+    # Demand bound
+    for ci in model.s_CI:
+        for t in model.s_T:
+            if t in model.s_T_ci[ci]:
+                model.v_Demand[ci, t].setub(model.p_ConsumerRate[ci])
+            else:
+                model.v_Demand[ci, t].setub(0)
+
+    """
+    # NOTE: Will need to reformulate the restriction set if it is desired in the bidding model; by going with a clearing model, we have sacrificed request-level control over shipping; we now work at the node-level, meaning we would need to block transport between nodes if a request is declined.
+    # restrict trucking from declined match
+    def TruckingUpperBound(model, pi, ci, t):
+        if ((model.p_ProducerOperator[pi],model.p_ProducerPad[pi], model.p_ConsumerOperator[ci], model.p_ConsumerPad[ci]) in restricted_set):
+            return model.v_FP_Trucked[pi, ci, t] == 0
+        else:
+            return Constraint.Skip
+    
+    model.TruckingUpperBound = Constraint(
+        model.s_PI,
+        model.s_CI,
+        model.s_T,
+        rule=TruckingUpperBound,
+        doc="Trucking Upper Bound"
+    )
+    """
+    print("All variable bounds set")
+
+    
+    ### ---------------------- Constraints ---------------------------------------------
+
+    # -------- base constraints ----------
+    # demand balance
+    def ConsumerDemandBalanceRule(model, c, t):
```

Comment:
Another general comment, we do not need to explicitly write bool to check the value of a parameter being greater than 0. Just writing `if param` would compare the boolean values and would give false if the param value if 0 or None

Reply:
Neat. Are they equivalent? Is there any difference between them I should be aware of?